### ISSUE TITLE
False positive in RET503 since ruff 0.0.248

### ISSUE THREAD

**Issue by link2xt** [charliermarsh/ruff#3071]
Since version 0.0.248 ruff flags this code as having an error:
```python
def foo():
    while True:
        return True
```

I have `git bisect`ed this down to commit 059601d9685098a071bac6149ef5f321b3d845b8, PR #2962.

Running `$ cargo run -p ruff_cli -- check ./x.py --no-cache --select RET503` produces this error:
```
x.py:2:5: RET503 [*] Missing explicit `return` at the end of function able to return non-`None` value
```

This `while True` is a common pattern in Python to make infinite loops, and it clearly never returns unless there is a `break` inside.
I do not want to `return None` in the end, because I annotate my functions with types, and having a `return None` at the end of a function that is supposed to return an object (in this case I can add `return False` in the end, but not all objects are as easily constructable) would be flagged by mypy/pyright.

**Comment by charliermarsh** [comment]
I guess we could treat `while True:` differently than other `while` loops. The issue is that, in general, we can't be certain that the loop will actually execute, so even if it ends in a `return`, we treat it as if it's non-returning. (The previous version just omitted `while` loops entirely, which is also a valid option and probably easier.)

**Comment by charliermarsh** [comment]
I'll make sure it's fixed for the next release.

**Comment by charliermarsh** [comment]
For now I just omitted `while True` entirely. We can do better of course but hopefully this avoids these false positives for you. (Previously, I think we ignored _all_ `while` loops.)
